sdk/cosmos/azure-cosmos/test/test_resource_id.py

```diff
@@ -97,23 +97,37 @@ def test_create_illegal_characters_async(self):
                 self.fail("Database create should have failed for id {}".format(resource_id))
             except ValueError as e:
                 assert str(e) in error_strings
+            # Let service throw size exception
+            except exceptions.CosmosHttpResponseError as e:
+                assert e.status_code == http_constants.StatusCodes.BAD_REQUEST
+                assert "Ensure to provide a unique non-empty string less than '255' characters." in e.message
 
             try:
                 created_database.create_container(id=resource_id, partition_key=partition_key)
                 self.fail("Container create should have failed for id {}".format(resource_id))
             except ValueError as e:
                 assert str(e) in error_strings
+            except exceptions.CosmosHttpResponseError as e:
+                assert e.status_code == http_constants.StatusCodes.BAD_REQUEST
+                assert "Ensure to provide a unique non-empty string less than '255' characters." in e.message
```

Comment:
should here be 1024 then?

Reply:
this assertion is 255 because that is the limit for containers/databases